quisby/pricing/cloud_pricing.py

```diff
@@ -117,40 +146,50 @@ def get_aws_instance_info(instance_name, region):
     return response['PriceList']
 
 
-def get_aws_pricing(instance_name, region):
-    price_list = get_aws_instance_info(instance_name, region)
+def get_aws_pricing(instance_type, region, os_type):
+    product = "AmazonEC2"
+    client = boto3.client('pricing', region_name=region)
+    filters = [
+        {"Type": "TERM_MATCH", "Field": "regionCode", "Value": region},
+        {"Type": "TERM_MATCH", "Field": "instanceType", "Value": instance_type},
+        {"Type": "TERM_MATCH", "Field": "operatingSystem", "Value": os_type},
+    ]
 
-    # Filter pricing details
-    if price_list:
-        price_item = json.loads(price_list[0])
-
-        terms = price_item["terms"]
-
-        price_dimension = terms["OnDemand"][iter(terms["OnDemand"]).__next__()][
-            "priceDimensions"
-        ]
-        price = price_dimension[iter(price_dimension).__next__()]["pricePerUnit"]["USD"]
-
-        return price
-    else:
-        return None
+    response = client.get_products(ServiceCode=product, Filters=filters)
 
+    if "PriceList" in response:
+        try:
+            price_list = json.loads(response["PriceList"][0])
+
+            # Extract pricing information as needed
+            terms = price_list["terms"]["OnDemand"]
+            for _, term_info in terms.items():
+                for _, price_dimension in term_info["priceDimensions"].items():
+                    price_per_hour = price_dimension["pricePerUnit"]["USD"]
+                    print(
```

Comment:
Only a question: do we want to leave this what looks like debugging line in?  I'm trying to decide from just looking at it if we'll get a bunch of output that might mask actual issues just by volume.

Reply:
Are you talking about line 169 ?The print statement ?

Reply:
Yes, that's where github *should* have left the comment... :)